doc/adr/0007-queryable-properties.md

```diff
@@ -0,0 +1,286 @@
+    end
+    User->>client: select "Go to crash time" menu item
+    client->>server: POST /experiment/{expUuid}/outputs/{outputId}/timegraph/properties/{propertyId}
+    Note over client,server: requested_items=[selected_entry_Id] representing thread
+    server->>client: PropertyModel ([ { time_range: {start}, metatdata: (cpu, 0) } ])
+    client->>client: broadcast selectionRange({ time_range:{start}, metatdata: (cpu, 0) })
+    client->>client: in all views, update selection range and show cpu (if applicable) 
+    User->>client: navigate to selectionRange {start}
+```
+
+#### Open source code
+
+Some UI elements, e.g. trace event in `Events Table` or state in `Thread Status` view, it can provide the source file and the line number where that event happened. Use this to allow provide callsite information and use the IDE's capability to open source files.
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant client as Theia Trace
+    participant server as Trace Server
+    Note over client,server: Prerequisite Open Trace and fetched property descriptors
+    User->>client: open Thread Status View
+    client->>server: POST /experiment/{expUuid}/outputs/{outputId}/table/columns
+    server->>client: 200: ColumnsModel
+    client->>client: populate tree
+    client->>server: POST /experiment/{expUuid}/outputs/{outputId}/table/lines
+    server->>client: 200: LineModel
+    client->>client: populate rows and states
+    User->>client: right-mouse click on line
+    alt one desc of propertyDescriptors needs metadata_keys
+        client->>server: get metadata_keys (if needed) *
+    end
+    loop while descriptor in propertyDescriptors
+        Note over client: Conditionally add menu item (metadata_keys)
+        client->>client: create menu entry (using descriptor.name, descriptor.description, descriptor.propertyId)
+    end
+    User->>client: select "Open source code" menu item
+    client->>server: POST /experiment/{expUuid}/outputs/{outputId}/timegraph/properties/{propertyId}
+    Note over client,server: requested_items=[selected_entry_Id], requested_table_index[index]
+    server->>client: PropertyModel ([ { callsite: {start, line} ])
+    client->>client: Open source code using IDE
+```
+
+Notes: * This doesn't exists yet and needs to be implemented. Tooltip endpoint can be added for that.
+
+## Decision
+
+Implement this feature for the `Go to min/max` for Latency views first which represents an example time navigation use case. Make it the TSP definition extendible so that the other use cases can be added gradually.
+
+## Consequences
+
+If this proposal is implemented, the tool becomes more interactive and it will provide a better UX. Users can retrieve more details from the backend in order to navigate inside the trace and find relevant, interesting time ranges, to look-up source code or create custom data providers.
```

Comment:
Can you clarify what you mean by "the tool"? You mean the frontend (i.e. theia-trace-extension) only?

Reply:
yes, it's the front-end, but not limited to this client. I think it's ok to be specific here because the theia-trace-extension fornt-end will be reference implementation for it.

Reply:
For some reasons I was imagining the [tsp-python-client] to be the reference implementation. I think it is much more intuitive, easy to understand and portable compared to the theia-trace-extension UI. However, independently on this, I think it will improve the understanding if you can update the word "tool" to "theia-trace-extension" in case you'll do a follow up commit.

[tsp-python-client]: https://github.com/eclipse-cdt-cloud/tsp-python-client